torch/_dynamo/guards.py

```diff
@@ -568,7 +568,7 @@ def TENSOR_MATCH(self, guard: Guard):
             if not static:
                 if hasattr(value, "_dynamo_dynamic_indices"):
                     code.append(
-                        f"({tensor_name}._dynamo_dynamic_indices.issubset({value._dynamo_dynamic_indices})) if hasattr({tensor_name}, '_dynamo_dynamic_indices') else True"  # noqa: B950
+                        f"(({tensor_name}._dynamo_dynamic_indices.issubset({value._dynamo_dynamic_indices})) if hasattr({tensor_name}, '_dynamo_dynamic_indices') else True)"  # noqa: B950
```

Comment:
Now that we have https://github.com/pytorch/pytorch/pull/99566 - I can followup on this PR by moving this into TensorGuards